From addd5d0dbea76a330fbce287d888d48c855d16c5 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:24:24 -0400
Subject: [cpufreq] intel_pstate: Remove unnecessary type casting in div_s64() call

Message-id: <1409235879-1283-171-git-send-email-prarit@redhat.com>
Patchwork-id: 89493
O-Subject: [RHEL7.1 PATCH BZ 1134639 170/185] cpufreq: intel_pstate: Remove unnecessary type casting in div_s64() call
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit fa30dff9a81ea9fdc2e985a14fe14ce6393a3214
Author: Stratos Karafotis <stratosk@semaphore.gr>
Date:   Fri Jul 18 08:37:18 2014 -0700

    cpufreq: intel_pstate: Remove unnecessary type casting in div_s64() call

    div_s64() accepts the divisor parameter as s32. Helper div_fp()
    also accepts divisor as int32_t.

    So, remove the unnecessary int64_t type casting.

    Signed-off-by: Stratos Karafotis <stratosk@semaphore.gr>
    Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 601c428..e5fd780 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -50,7 +50,7 @@ static inline int32_t mul_fp(int32_t x, int32_t y)
 
 static inline int32_t div_fp(int32_t x, int32_t y)
 {
- return div_s64((int64_t)x << FRAC_BITS, (int64_t)y);
+ return div_s64((int64_t)x << FRAC_BITS, y);
 }
 
 struct sample {
-- 
1.7.1